### PR TITLE
feat(across-relayer): enable batch disputing

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

The disputer should be able to send batches of disputes. This PR enables this.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/insured-bridge-relayer/src/Relayer.ts [hunk 1]
```diff
@@ -123,21 +123,30 @@ export class Relayer {
...
       return;
     }
-    this.logger.debug({ at: "AcrossRelayer#Disputer", message: `Processing ${pendingRelays.length} pending relays` });
-
-    for (const relay of pendingRelays) {
+    this.logger.debug({
+      at: "AcrossRelayer#Disputer",
+      message: `Processing ${Object.keys(pendingRelays).length} l1 token pending relays`,
+    });
+    for (const l1Token of Object.keys(pendingRelays)) {
```
Comment (chrismaree) [inline packages/insured-bridge-relayer/src/Relayer.ts:135]: this change (and the introduction of `getPendingRelayedDepositsGroupedByL1Token`) is done to ensure that we only send the batch disputes to one contract _at a time_. i.e consider if both the USDC and WETH pool have disputes. we cant send these in a batch as the destination contract is different between the two. The implementation here now sends them based off a grouping of `BridgePool`.

File: packages/insured-bridge-relayer/src/Relayer.ts [hunk 2]
```diff
@@ -496,23 +499,12 @@ export class Relayer {
...
-    if (receipt?.events?.RelayCanceled)
-      this.logger.error({
-        at: "AcrossRelayer#Disputer",
-        message: "Dispute failed to send to OO 👀!",
-        tx: receipt.transactionHash,
-        depositHash: receipt.events.RelayCanceled.returnValues.depositHash,
-        relayHash: receipt.events.RelayCanceled.returnValues.relayHash,
-        disputer: receipt.events.RelayCanceled.returnValues.disputer,
-      });
+  private async _generateDisputeRelayTransaction(deposit: Deposit, relay: Relay) {
```
Comment (chrismaree) [inline packages/insured-bridge-relayer/src/Relayer.ts:502]: this method now returns the dispute transaction, just like the similar methods in this file `_generateRelayTransaction` and `_generateSettleTransactionForSettleableRelay`

File: packages/insured-bridge-relayer/src/Relayer.ts (hunk 1 above)
Comment (nicholaspai) [reply packages/insured-bridge-relayer/src/Relayer.ts:135]: You should add this as a comment to the code to make sure no one accidentally introduces the possiiblity that batches can be sent to multiple contracts

**Review by nicholaspai — APPROVED** [review]
Looks great, LGTM

**Pull request merged**